### PR TITLE
Wrap Aggregate API

### PR DESCRIPTION
**Pull request by nguyenv**
Wrap Aggregate API (x-ref https://github.com/TileDB-Inc/TileDB/pull/4438).

### CODE REVIEW

File: tiledb/libtiledb.pyx [hunk 1]
```diff
@@ -1701,6 +1702,103 @@ cdef class Array(object):
...
+
+            if isinstance(cond, str):
+                q.set_cond(QueryCondition(cond))
+            elif isinstance(cond, QueryCondition):
+                raise TileDBError(
+                    "Passing `tiledb.QueryCondition` to `cond` is no longer "
+                    "supported as of 0.19.0. Instead of "
+                    "`cond=tiledb.QueryCondition('expression')` "
+                    "you must use `cond='expression'`. This message will be "
+                    "removed in 0.21.0.",
```
Comment (nguyenv) [inline tiledb/libtiledb.pyx:1763]: These will be removed in a separate PR.

**Review by Shelnutt2 — APPROVED** [review]
